controller/hybridgateway/controller.go

```diff
@@ -224,8 +226,14 @@ func (r *HybridGatewayReconciler[t, tPtr]) Reconcile(ctx context.Context, req ct
 		return ctrl.Result{Requeue: true}, nil
 	}
 
+	// If translation is not finished and needs requeue, do it now.
+	if needsRequeue {
```

Comment:
The `needsRequeue` value is returned by the translate function, but it is checked after we have enforced the state by calling `enforceState`. At this point, I would make the `needsRequeue` variable and internal state of the converter implementation and check it in the `enforceState` function which already returns a values that is used re-queue the reconciled object.

Reply:
Are you suggesting to extend the Converter interface with some sort of "GetNeedsRequeue()" function? Otherwise how would you get access to an implementation specific state? 

Re-looking at the code seems we may get rid of the needsRequeue as the enforceState will *always* update or create something if the Translate produced something anew. Not on the super-safe side as we implicitly rely on the flow of the code, but cannot image any case in which this would not work.